### PR TITLE
chore: Upgrade test dependency versions

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

**Comment by czp13** [comment]
Error:
```
Failed to execute goal on project vaadin-testbench-unit-shared: Could not resolve dependencies for project com.vaadin:vaadin-testbench-unit-shared:jar:9.0-SNAPSHOT: Could not find artifact org.springframework:spring-test:jar:6.1.0 in vaadin-prereleases (https://maven.vaadin.com/vaadin-prereleases/)
```
6.0.9 seems the latest for this jar:
https://mvnrepository.com/artifact/org.springframework/spring-test

Also for Spring:
https://github.com/spring-projects/spring-framework/releases

File: pom.xml [hunk 1]
```diff
@@ -36,11 +36,11 @@
         <selenium.version>4.8.3</selenium.version>
         <maven.compiler.source>17</maven.compiler.source>
         <maven.compiler.target>17</maven.compiler.target>
-        <spring.version>6.0.0-M6</spring.version>
-        <spring.security.version>6.0.0-M7</spring.security.version>
-        <slf4j.version>2.0.3</slf4j.version>
-        <failsafe.version>3.0.0-M7</failsafe.version>
-        <surefire.version>3.0.0-M7</surefire.version>
+        <spring.version>6.1.0</spring.version>
```
Comment (czp13) [inline pom.xml:39]: Maybe?
```suggestion
        <spring.version>6.0.9</spring.version>
```

File: pom.xml [hunk 2]
```diff
@@ -36,11 +36,11 @@
...
+        <spring.security.version>6.1.0</spring.security.version>
```
Comment (czp13) [inline pom.xml:40]: security has 6.1.0 interestingly so maybe we can let it be 6.1.0.

**Comment by Artur-** [comment]
Yes, Spring Boot 3.1.0 indeed uses Spring Security 6.1.0 and Spring Framework 6.0.9

**Review by czp13 — APPROVED** [review]

**Pull request merged**